<?xml version="1.0"?>
<robot name="crane3r">
  <!-- Links -->
  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://kine/meshes/Crane3R/4200.obj"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 2.1" rpy="0 0 0"/>
        <geometry>
            <cylinder length="4.2" radius="0.5"/>
        </geometry>
    </collision>
  </link>
  <link name="boom_link">
    <visual>
      <geometry>
        <mesh filename="package://kine/meshes/Crane3R/7000.obj"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="3.5 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="7.0 1 1"/>
      </geometry>
    </collision>
  </link>
  <link name="jib_link">
    <visual>
      <geometry>
        <mesh filename="package://kine/meshes/Crane3R/5200.obj"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="2.6 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="5.2 1 1"/>
      </geometry>
    </collision>
  </link>
  <link name="end_effector"/>

  <!-- base_joint: rotates about Z between world -> base_link -->
  <joint name="base_joint" type="revolute">
    <parent link="world"/>
    <child link="base_link"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit lower="-3.1416" upper="3.1416" effort="1.0" velocity="1.0"/>
  </joint>

  <!-- boom_joint: offset (0,0,4.2), rotates about Y (pitch) base_link -> boom_link -->
  <joint name="boom_joint" type="revolute">
    <parent link="base_link"/>
    <child link="boom_link"/>
    <origin xyz="0 0 4.2" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit lower="-1.396" upper="0" effort="1.0" velocity="1.0"/>
  </joint>

  <!-- jib_joint: offset (7,0,0), rotates about Y (pitch) boom_link -> jib_link -->
  <joint name="jib_joint" type="revolute">
    <parent link="boom_link"/>
    <child link="jib_link"/>
    <origin xyz="7 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit lower="0.69" upper="2.44" effort="1.0" velocity="1.0"/>
  </joint>

  <!-- end effector fixed offset (5.2,0,0) -->
  <joint name="ee_fixed" type="fixed">
    <parent link="jib_link"/>
    <child link="end_effector"/>
    <origin xyz="5.2 0 0" rpy="0 0 0"/>
  </joint>

  <link name="base"/>
  <joint name="base_link-base" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="base"/>
  </joint>
</robot>